server/map_rpc_server.go

```diff
@@ -49,11 +49,85 @@ func (t *TrillianMapServer) IsHealthy() error {
 	return t.registry.MapStorage.CheckDatabaseAccessible(context.Background())
 }
 
+// Init creates the initial revision 0 SignedMapHead, if one doesn't already exist.
+func (t *TrillianMapServer) Init(ctx context.Context, mapID int64) error {
+	tree, hasher, err := t.getTreeAndHasher(ctx, mapID, false /* readonly */)
+	if err != nil {
+		return err
+	}
+	ctx = trees.NewContext(ctx, tree)
+
+	tx, err := t.registry.MapStorage.BeginForTree(ctx, mapID)
+	if err != nil && err != storage.ErrMapNeedsInit {
+			return err
+	}
+	defer tx.Close()
+
+	if err == nil {
+		// Init() not needed.
+		return nil
+	}
+
+	glog.V(2).Infof("%v: Need to init map root revision 0", mapID)
+
+	// TODO(phad): Refactor the SetLeaves func to avoid the duplication that follows.
+	smtWriter, err := merkle.NewSparseMerkleTreeWriter(
+		ctx,
+		mapID,
+		0 /* write revision */,
+		hasher, func() (storage.TreeTX, error) {
+			ttx, err := t.registry.MapStorage.BeginForTree(ctx, mapID)
+			if err == storage.ErrMapNeedsInit {
+				err = nil // Init() is ongoing, so ignore this.
+			}
+			return ttx, err
+		})
+	if err != nil {
+		return err
+	}
+
+	rootHash, err := smtWriter.CalculateRoot()
+	if err != nil {
+		return fmt.Errorf("CalculateRoot(): %v", err)
+	}
+	newRoot := trillian.SignedMapRoot{
+		TimestampNanos: time.Now().UnixNano(),  // TODO(phad): should this be the tree create timestamp?
```

Comment:
This is correct.  We want to timestamp when each SignedMapRoot is created. 
If we were to create this SignedMapRoot during admin.NewTree, then the timestamps would match.